src/NewTools-Debugger/StDebuggerSettings.class.st

```diff
@@ -0,0 +1,128 @@
+"
+A `StDebuggerSettings` class holds and manages global settings for the Pharo Debugger.
+
+This class provides a centralized way to configure and access various debugger settings that affect the behavior and appearance of the Pharo Debugger across the system.
+
+## Responsibility
+- Store and provide access to global debugger settings
+- Offer methods to modify these settings
+- Ensure settings persistence across sessions
+
+## Collaborators
+- StDebugger: The main debugger class that uses these settings
+- Settings Framework: For integration with Pharo's settings browser
+
+## Examples
+```
+    StDebuggerSettings lastKnownPosition.
+    StDebuggerSettings preserveWindowPosition.
+```
+
+## Internal Representation and Key Implementation Points:
+- Uses class-side methods and class variables to store settings
+- Integrates with Pharo's Settings framework for easy configuration through the Settings Browser
+
+Note: This class is part of the core debugging infrastructure in Pharo and should be modified with caution.
+
+
+"
+Class {
+	#name : 'StDebuggerSettings',
+	#superclass : 'Object',
+	#classVars : [
+		'LastKnownExtent',
+		'LastKnownPosition',
+		'PreserveWindowPosition'
+	],
+	#category : 'NewTools-Debugger-View',
+	#package : 'NewTools-Debugger',
+	#tag : 'View'
+}
+
+{ #category : 'accessing' }
+StDebuggerSettings class >> defaultExtent [
+
+	^ (850 @ 650) scaledByDisplayScaleFactor
+
+]
+
+{ #category : 'accessing' }
+StDebuggerSettings class >> defaultPosition [
+
+	^ 10 @ 10
+
+]
+
+{ #category : 'class initialization' }
+StDebuggerSettings class >> initialize [
+
+	LastKnownPosition := self defaultPosition.
+	LastKnownExtent := self defaultExtent.
+]
+
+{ #category : 'initialization' }
+StDebuggerSettings class >> initializeSettings: aWindowPresenter [
+
+	self preserveWindowPosition ifFalse: [ ^ self ].
+	aWindowPresenter window ifNil: [ ^ self ].
+	aWindowPresenter whenClosedDo: [
+		self
+			lastKnownPosition: aWindowPresenter window position;
+			lastKnownExtent: aWindowPresenter window extent ]
+]
```

Comment:
this method should not be called "initializeSettings:" but rather "configure debugger window" or something similar.

Reply:
For compatibility with Spec 2, I renamed it as `initializeWindow:`

Reply:
I don't know if it is an initialization really, as the usage is in the end to capture in the settings the last known position of the window.

I also think that there should not be any compatbility with spec, perhaps settings should have an explicit "setting" api.

I didn't catch it at start, but any chance of getting rid of the nil check?
Is it possible that a closing window presenter has no window?
I know sometimes spec does strange things and we can't do anything but I like to try.

Reply:
thanks for the hard work :)